Toplean_Alexandru/week-02/Domain/Domain/CreateMenuOp/CreateMenuCmd.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Domain.Models;
+
+namespace Domain.Domain.CreateMenuOp
+{
+    public struct CreateMenuCmd
+    {
+        public Restaurant Restaurant { get; }
+        public string Name { get; }
+        public MenuType MenuType { get; }
+
+        public CreateMenuCmd(Restaurant restaurant, string name, MenuType menuType)
+        {
+            Restaurant = restaurant;
+            Name = name;
+            MenuType = menuType;
+        }
+
+        // Returns a tuple. If everything is valid, the function will return true and the MenuErrorCode will be None.
+        // If it is not valid, then the function will return false with the known reason.
+        // Try-Catch block also added in case there is an unknown error and it will return false.
+        public (bool, MenuErrorCode) IsValid()
+        {
+            try
+            {
+                return HasIllegalCharacters(Name) ? (false, MenuErrorCode.IllegalCharacter) :
+                    NameTooShort(Name) ? (false, MenuErrorCode.EmptyField) :
+                    Restaurant == null ? (false, MenuErrorCode.NullRestaurant) :
+                    (true, MenuErrorCode.None);
+            }
+            catch
+            {
+                return (false, MenuErrorCode.UnknownError);
+            }
+        }
+
+        public bool NameTooShort(String name) => name.Length > 0 ? false : true;
+
+        public bool HasIllegalCharacters(String name)
+        {
+            foreach (char c in name)
```

Comment:
Try to rewrite this check using Any LINQ operator: https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.any?view=netframework-4.8